### ISSUE TITLE
zu.text to return array, not joined string

### ISSUE THREAD

**Issue by algesten** [algesten/zu#4]
After using this in a coupe of projects, there are two things that bug me about the API. This is one of them:

`zu.text(nodes)` returns one big joined text out of the given node list.

Mostly I use zu to pick out data from an XML document. Example

``` xml
<npexchange>
  <article someattr="blaha"><article_part_type_id>art1</article_part_type_id></article>
  <article someattr="foho"><article_part_type_id>art1</article_part_type_id></article>
  <article someattr="bla2"><article_part_type_id>art1</article_part_type_id></article>
</npexchange>
```

If I want to get the article part ids of this document, it would be reasonable to expect:

``` javascript
zu.text(zu.find(xml, "article > article_part_type_id"));   //=> NOT ['art1', 'art2', 'art3']
```

But instead we get `art1art2art3`. This is because `zu.text` is analogous to `zu.xml` and as a last step does an `arr.join('')` to merge all text bits into one string. The solution is currently (which has tripped both me an my team members up):

``` javascript
zu.find(xml, "article > article_part_type_id").map(zu.text);   //=> ['art1', 'art2', 'art3']
```

It's not a terrible solution, but on some level I feel that if I really want to join all text bits together to one string, I can easily do that `arr.join('')` myself. The reason it is like this is likely because we may want to get text out like:

``` html
<p>It was a dark and <b>stormy</b> night; the rain fell in torrents.</p>
```

If I `zu.text` the above, without the automatic `join`, I would get a three element list.

To sum it up: The current `zu.text` only makes sense if we are mostly working with formatted text – which isn't my typical use case. I don't feel it's a terrible crime to break the analogy with `zu.xml`, since zu.xml actually _preserves_ the structure – albeit it turns the array values into tag wrapped data – we could go backwards from zu.xml back to arrays.. `zu.text` is on the other hand destructive.
